app/grandchallenge/core/error_handlers.py

```diff
@@ -0,0 +1,98 @@
+from grandchallenge.notifications.models import Notification, NotificationType
+
+
+class ErrorHandler:
+    def handle_error(self, *, error_message, user=None, interface=None):
+        raise NotImplementedError("Subclasses must implement this method")
+
+
+class JobCIVErrorHandler(ErrorHandler):

```

Comment:
Some of the names are still unclear to me, could you add some docstrings about which situations each error handler should be used?